spec/outputs/s3_spec.rb

```diff
@@ -3,4 +3,238 @@
 require 'logstash/outputs/s3'
 
 describe LogStash::Outputs::S3 do
+  before do
+    # We stub all the calls from S3, for more information see:
+    # http://ruby.awsblog.com/post/Tx2SU6TYJWQQLC3/Stubbing-AWS-Responses
+    AWS.stub!
+  end
+
+  let(:minimal_settings)  {  { "access_key_id" => "1234",
+                               "secret_access_key" => "secret",
+                               "bucket" => "my-bucket" } }
+
+  describe "configuration" do
+    it "should support the deprecated endpoint_region as a configuration option" do
+      config = { "endpoint_region" => "sa-east-1" }
+      s3 = LogStash::Outputs::S3.new(config)
+      expect(s3.aws_options_hash[:s3_endpoint]).to eq("s3-sa-east-1.amazonaws.com")
+    end
+
+    it "should use the depracated option before failling back to the region" do
+      config = { "region" => "us-east-1", "endpoint_region" => "sa-east-1" }
+      s3 = LogStash::Outputs::S3.new(config)
+      expect(s3.aws_options_hash[:s3_endpoint]).to eq("s3-sa-east-1.amazonaws.com")
```

Comment:
To check that you can use this matcher http://www.relishapp.com/rspec/rspec-expectations/v/2-14/docs/built-in-matchers/include-matcher#hash-usage